### DISCUSSION TITLE
Please Add Tongou TOWSMR1 1-40

_TZE204_kobbcyum

### DISCUSSION THREAD

**Discussion by terzo33** [Koenkk/zigbee2mqtt#22826] [Devices]
Hi, please add this Smart circuit breaker zigbee. Thanks a lot

Tongou TOWSMR1 1-40

_TZE204_kobbcyum

![Screenshot_20240509_174104_AliExpress.jpg](https://github.com/Koenkk/zigbee-herdsman-converters/assets/71357473/e1eec95d-2583-46fd-aeee-ae00a071184d)

How can I help to add to zigbee2mqtt?

![Screenshot_20240509_201819_Home Assistant.jpg](https://github.com/Koenkk/zigbee-herdsman-converters/assets/71357473/a4eae82b-3dcc-4d4b-ae67-e247cc613438)

```js

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const modernExtend = require('zigbee-herdsman-converters/lib/modernExtend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('zigbee-herdsman-converters/lib/tuya');

const definition = {
    // Since a lot of TuYa devices use the same modelID, but use different datapoints
    // it's necessary to provide a fingerprint instead of a zigbeeModel

fingerprint: [
        {
            // The model ID from: Device with modelID 'TS0601' is not supported
            // You may need to add \u0000 at the end of the name in some cases
            modelID: 'TS0601',
            // The manufacturer name from: Device with modelID 'TS0601' is not supported.
            manufacturerName: '_TZE204_kobbcyum',
        },
    ],
    model: 'TOWSMR1',
    vendor: 'Tongou',
    description: 'Smart circuit breaker',
    fromZigbee: [tuya.fz.datapoints],
    toZigbee: [tuya.tz.datapoints],
    onEvent: tuya.onEventSetTime, // Add this if you are getting no converter for 'commandMcuSyncTime'
    configure: tuya.configureMagicPacket,
    exposes: [tuya.exposes.switch(), e.energy(), e.temperature(), e.power(), e.voltage(), e.current(),
        exposes.enum('fault', ea.STATE, ['clear', 'over_current_threshold', 'over_power_threshold',
            'over_voltage threshold', 'wrong_frequency_threshold']).withDescription('Fault status of the device (clear = nothing)'),
        exposes.enum('threshold_1', ea.STATE, ['not_set', 'over_current_threshold', 'over_voltage_threshold'])
            .withDescription('State of threshold_1'),
        exposes.binary('threshold_1_protection', ea.STATE, 'ON', 'OFF')
            .withDescription('OFF - alarm only, ON - relay will be off when threshold reached'),
        exposes.numeric('threshold_1_value', ea.STATE)
            .withDescription('Can be in Volt or Ampere depending on threshold setting. Setup the value on the device'),
        exposes.enum('threshold_2', ea.STATE, ['not_set', 'over_current_threshold', 'over_voltage_threshold'])
            .withDescription('State of threshold_2'),
        exposes.binary('threshold_2_protection', ea.STATE, 'ON', 'OFF')
            .withDescription('OFF - alarm only, ON - relay will be off when threshold reached'),
        exposes.numeric('threshold_2_value', ea.STATE)
            .withDescription('Setup value on the device'),
        exposes.binary('clear_fault', ea.STATE_SET, 'ON', 'OFF')
            .withDescription('Turn ON to clear last the fault'),
        exposes.text('meter_id', ea.STATE).withDescription('Meter ID (ID of device)'),
    ],
    meta: {
        tuyaDatapoints: [
            [1, 'energy', tuya.valueConverter.divideBy100],
            [3, null, null], // Monthly, but sends data only after request
            [4, null, null], // Dayly, but sends data only after request
            [6, null, tuya.valueConverter.phaseVariant2], // voltage and current
            [10, 'fault', tuya.valueConverterBasic.lookup({'clear': 0, 'over_current_threshold': 1,
                'over_power_threshold': 2, 'over_voltage_threshold': 4, 'wrong_frequency_threshold': 8})],
            [11, null, null], // Frozen - strange function, in native app - nothing is clear
            [16, 'state', tuya.valueConverter.onOff],
            [17, null, tuya.valueConverter.threshold], // It's settable, but can't write converter
            [18, 'meter_id', tuya.valueConverter.raw],
            [20, 'clear_fault', tuya.valueConverter.onOff], // Clear fault
            [21, null, null], // Forward Energy T1 - don't know what this
            [22, null, null], // Forward Energy T2 - don't know what this
            [23, null, null], // Forward Energy T3 - don't know what this
            [24, null, null], // Forward Energy T4 - don't know what this
        ],
    },
};

module.exports = definition;

```

with this configuration it partially works, i.e. some sensors are missing and those that work are updated every 5/10 minutes

![Screenshot_20240510_072649_Home Assistant.jpg](https://github.com/Koenkk/zigbee-herdsman-converters/assets/71357473/7d65ef50-98bb-45bf-9120-945ebcf5c7cd)

![Screenshot_20240510_072150_Home Assistant.jpg](https://github.com/Koenkk/zigbee-herdsman-converters/assets/71357473/c6e53322-16f8-4f80-b172-56e91232dabb)

![Screenshot_20240510_072206_Home Assistant.jpg](https://github.com/Koenkk/zigbee-herdsman-converters/assets/71357473/5a422c0a-6c56-4e48-9b5e-61470034d44b)

the missing sensors are:

Temperature threshold
Temperature breaker
Power threshold
Power breaker
Over current threshold
Over current breaker
Over voltage threshold
Over voltage breaker
Under voltage threshold
Under voltage breaker
Power outage memory
Indicator mode

Thanks

**Comment by jordiinstalaciones** [comment]
I am also interested in having compatibility for these devices as I have 2 of them but I am unable to do anything with them. Thank you

**Comment by meremortals70** [comment]
Another vote for this please?

**Comment by Imanol82** [comment]
I'm interested too. Thank you very much.

I've installed the dev but not all appears.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/109697268/8b24a58f-8682-45cc-8869-af3e531710ba)

The state is missing.

**Comment by terzo33** [comment]
Hi @Koenkk , after update z2m to 1.3.9 some sensors are missing and those sensors that work are updated every 5/10 minutes

![Screenshot_20240702_073442_Home Assistant.jpg](https://github.com/Koenkk/zigbee2mqtt/assets/71357473/6e143100-21de-49df-afff-25763f17e8a7)

the missing sensors are:

Temperature threshold
Temperature breaker
Power threshold
Power breaker
Over current threshold
Over current breaker
Over voltage threshold
Over voltage breaker
Under voltage threshold
Under voltage breaker
Power outage memory
Indicator mode
Rate current Value
Earth Leakage Value

![Screenshot_20240702_081317_Home Assistant.jpg](https://github.com/Koenkk/zigbee2mqtt/assets/71357473/dfb40c04-5a32-43d9-b0fe-9e25df9b2055)


Thanks

**Comment by Imanol82** [comment]
The most inportant that I can't see is the State relay. Any idea?

![image](https://github.com/Koenkk/zigbee2mqtt/assets/109697268/e912d8bd-fcab-4f9f-a6ea-99f3aba57ef1)

![image](https://github.com/Koenkk/zigbee2mqtt/assets/109697268/3cd5586c-3997-4129-a926-a4b2b2205649)

Thank you very much!

**Comment by maiandrei** [comment]
Your "_TZE204_kobbcyum" seems to be supported now. 
If not already solved, [here's a little help](https://gist.github.com/Koenkk/bfd4c3d1725a2cccacc11d6ba51008ba?permalink_comment_id=5748753#gistcomment-5748753):
